components/omega/src/infra/IOStream.cpp

```diff
@@ -543,10 +513,81 @@ int IOStream::create(const std::string &StreamName, //< [in] name of stream
       if (Err != 0) {
          LOG_ERROR("Error attaching alarm to model clock for stream {}",
                    StreamName);
-         return Err;
+         return Fail;
       }
    }
 
+   // Check the file frequency to see if there will be multiple records or
+   // time slices per file
+   NewStream->Multiframe = false;
+   int FileFreq          = 0;
+   std::string FileUnitStr;
+
+   int Err1            = StreamConfig.get("FileFreq", FileFreq);
+   int Err2            = StreamConfig.get("FileFreqUnits", FileUnitStr);
+   TimeUnits FileUnits = TimeUnitsFromString(FileUnitStr);
+
+   // Various error checks
+   if (Err1 == 0 or Err2 == 0) {
```

Comment:
It may be helpful to include a log statement specifying the frequency (`Freq` and `FreqUnits`) and file frequency (`FileFreq` and `FileFreqUnits`) utilized by Omega.

Reply:
In general, we need to echo options to the log file, but I need to make some changes to the logging facility as part of the error handler so will make the changes then.

Reply:
Sounds good. Thanks.